tests/ui/impl-trait/precise-capturing/capturing-implicit.rs

```diff
@@ -6,9 +6,8 @@
 #![rustc_variance_of_opaques]
 
 fn foo(x: &()) -> impl IntoIterator<Item = impl Sized> + use<> {
-    //~^ ERROR ['_: o]
-    //~| ERROR ['_: o]
-    //~| ERROR `impl Trait` captures lifetime parameter
+    //~^ ERROR []
```

Comment:
Ah, this doesn't seem right. Or at least, it changes the semantics of what we mean by "capture all in-scope parameters" for nested opaques.

Without having yet looked at the code, I assume what's happening here is that the nested opaque doesn't see any nested lifetimes because of the `use<>` from the outer opaque. Is that right?

Reply:
I'm not actually certain if https://rust-lang.github.io/rfcs/3617-precise-capturing.html#summary prescribes what should happen here 🤔 -- it doesn't seem to do so, after having done a brief scan.

Reply:
I changed the code to recurse and capture all in-scope lifetimes when we have nested opaques.

Reply:
@compiler-errors I have re-implemented the resolution to walk the scopes and not stop at the closest opaque. Is this closer to what you had in mind?

Reply:
Yeah, I believe so. If in the future we want opaque type captures to stop at the closest opaque, we can do so; that actually seems like the right behavior, but this should mean there are no *observable* changes.